tracks/csharp/exercises/nucleotide-count/mentoring.md

```diff
@@ -0,0 +1,57 @@
+### Reasonable solutions
+
+### Using a foreach loop
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public static class NucleotideCount
+{
+    public static IDictionary<char, int> Count(string sequence)
+    {
+        var result = new Dictionary<char, int> { {'A', 0}, {'C', 0}, {'G', 0}, {'T', 0} };
+        
+        foreach (var c in sequence)
+        {
+            if (result.ContainsKey(c)) result[c]++;
+            else throw new ArgumentException("Invalid Nucleotide Symbol");
+        }
+
+        return result;
+    }
+}
+```
+
+### Foreach with ToDictionary
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public static class NucleotideCount
+{
+    private static readonly char[] ValidNucleotideSymbols = {'A', 'C', 'G', 'T'};
+    
+    public static IDictionary<char, int> Count(string sequence)
+    {
+        var result = ValidNucleotideSymbols.ToDictionary(c => c, c => 0);
+        
+        foreach (var c in sequence)
+        {
+            if (ValidNucleotideSymbols.Contains(c)) result[c]++;
+            else throw new ArgumentException("Invalid Nucleotide Symbol");
+        }
+
+        return result;
+    }
+}
+```
+
+### Common Suggestions
+
+ * Iterating over `sequence` more than once wastes cycles, memory, etc. Often it isn't obvious that this is happening because methods like `ToList`, `GroupBy` or `OrderBy` will iterate a list without the student having to write a loop.
```

Comment:
```suggestion
 * Iterating over `sequence` more than once wastes cycles, memory, etc. Often it isn't obvious that this is happening because methods like `ToList`, `GroupBy` or `Count` will iterate a list without the student having to write a loop.
```

I think `Count` is something that will be used more often than `OrderBy`.

Reply:
Good point, will change

Reply:
Resolved in 44237bb40e87603671e8911323c2bf9b5fae5173